launch/navigation.launch.py

```diff
@@ -3,65 +3,153 @@
-from launch import LaunchDescription
-from launch.actions import DeclareLaunchArgument
-from launch.actions import IncludeLaunchDescription, GroupAction
+from launch import LaunchDescription, LaunchContext
+from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription, OpaqueFunction, GroupAction, LogInfo
 from launch.launch_description_sources import PythonLaunchDescriptionSource
 from launch.substitutions import LaunchConfiguration, PythonExpression
 from launch_ros.actions import Node, PushRosNamespace
 from launch.conditions import IfCondition
+"""
+Description:
 
-MY_NEO_ROBOT = os.environ.get('MY_ROBOT', "mpo_700")
-MY_NEO_ENVIRONMENT = os.environ.get('MAP_NAME', "neo_workshop")
+You can launch this file using the following terminal commands:
+
+This launch file is used to start the navigation for simulated Neobotix robot. 
+It is launched after launching the "simulation.launch.py" file.
+
+You can launch this file using the following terminal commands:
+
+1. `ros2 launch neo_simulation2 navigation.launch.py --show-args`
+   This command shows the arguments that can be passed to the launch file.
+2. `ros2 launch neo_simulation2 navigation.launch.py my_robot:=mp_500 map_name:=neo_track1 use_sim_time:=True use_multi_robots:=False use_amcl:=False`
+   This command launches the simulation with sample values for the arguments.
+   !(case is important for True/False)
+
+"""
+
+# OpaqueFunction is used to perform setup actions during launch through a Python function
+def launch_setup(context: LaunchContext, my_neo_robot_arg, my_neo_env_arg, use_sim_time_arg, use_amcl_arg, use_multi_robots_arg, namespace_arg):
+    
+    my_neo_robot = my_neo_robot_arg.perform(context)
+    my_neo_environment = my_neo_env_arg.perform(context)
+    use_sim_time = use_sim_time_arg.perform(context)
+    use_amcl = use_amcl_arg.perform(context)
+    use_multi_robots = use_multi_robots_arg.perform(context)
+    namespace = namespace_arg.perform(context)
+
+    launch_actions = []
 
-def generate_launch_description():
-    use_multi_robots = LaunchConfiguration('use_multi_robots', default='False')
-    use_amcl = LaunchConfiguration('use_amcl', default='False')
-    use_sim_time = LaunchConfiguration('use_sim_time', default='True')
-    namespace = LaunchConfiguration('namespace', default='')
-    use_namespace = LaunchConfiguration('use_namespace', default='False')
     map_dir = LaunchConfiguration(
         'map',
         default=os.path.join(
             get_package_share_directory('neo_simulation2'),
             'maps',
-            MY_NEO_ENVIRONMENT+'.yaml'))
+            my_neo_environment+'.yaml'))
 
     param_file_name = 'navigation.yaml'
     param_dir = LaunchConfiguration(
         'params_file',
         default=os.path.join(
             get_package_share_directory('neo_simulation2'),
-            'configs/'+MY_NEO_ROBOT,
+            'configs/'+my_neo_robot,
             param_file_name))
-
+    
```

Comment:
remove additional indents

Reply:
the indentation seems to be fine in my code editor anna...i removed 2 empty lines but apart from that where do u feel it is wrong?